01_fib/harirakr/main_test.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNegativeFib(t *testing.T) {
+
+	assert.Equal(t, []int{0, 1, -1, 2, -3, 5, -8, 13, -21, 34}, fib(-10))
+	assert.Equal(t, []int{0, 1, -1}, fib(-3))
+	assert.Equal(t, []int{0, 1}, fib(-2))
+	assert.Equal(t, []int{0}, fib(-1))
+
+}
+
+func TestFib(t *testing.T) {
+
+	assert.Equal(t,
+		[]int{0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181},
+		fib(20))
+	assert.Equal(t, []int{0, 1, 1, 2, 3, 5, 8, 13, 21, 34}, fib(10))
+	assert.Equal(t, []int{0, 1}, fib(2))
+	assert.Equal(t, []int{0}, fib(1))
+	assert.Equal(t, []int{}, fib(0))
+
+}
+
+func TestMain(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := strconv.Quote("[0 1 1 2 3 5 8]\n")
+	actual := strconv.Quote(buf.String())
+	if expected != actual {
+		t.Errorf("main fail; expected=%v, actual=%v", expected, actual)
+	}
+}
+
+// Test input out of bounds
+func TestFibOutOfBounds(t *testing.T) {
+	var buf1 bytes.Buffer
+	out = &buf1
+
+	fib(93)
```

Comment:
you need to use int64 for fib calculations if you test bounds explicitly. int defaults to int32 on go playground for instance.

Reply:
Done